### PR TITLE
Allow to modify env for gdbserver process

### PR DESCRIPTION
**Pull request by arekzaluski**
## Changes

- add `gdbServerEnv` option. It allows to modify env for gdb server process.

### CODE REVIEW

File: src/abstract-server.ts [hunk 1]
```diff
@@ -92,6 +94,10 @@ export abstract class AbstractServer extends EventEmitter {
         return serverArguments || [];
     }
 
+    private resolveServerEnv(serverEnv?: NodeJS.ProcessEnv): NodeJS.ProcessEnv {
+        return serverEnv || nodeProcess.env;
```
Comment (thegecko) [inline src/abstract-server.ts:98]: IMO, this should be a shallow merge, like...

```
return {
  ...nodeProcess.env,
  ...serverEnv
};
```

Comment (arekzaluski) [reply src/abstract-server.ts:98]: I was thinking about it but was not convinced that it is the best idea. I can however change to it if you prefer.

Comment (arekzaluski) [reply src/abstract-server.ts:98]: Removing all env variables would be impossible using this solution (unless client list each of them).

Comment (thegecko) [reply src/abstract-server.ts:98]: do you want to remove them all?

Comment (arekzaluski) [reply src/abstract-server.ts:98]: yes

**Pull request merged**